pkg/watcher/conditionwatcher.go

```diff
@@ -61,18 +61,18 @@ type waitEntry struct {
 	waitChan      chan waitResult
 }
 
-type conditionWatcher struct {
+type ConditionWatcher struct {
 	informerFactory azurediskInformers.SharedInformerFactory
 	waitMap         sync.Map // maps namespaced name to waitEntry
 	namespace       string
 }
 
-func newConditionWatcher(ctx context.Context, azDiskClient azDiskClientSet.Interface, informerFactory azurediskInformers.SharedInformerFactory, namespace string) *conditionWatcher {
+func NewConditionWatcher(ctx context.Context, azDiskClient azDiskClientSet.Interface, informerFactory azurediskInformers.SharedInformerFactory, namespace string) *ConditionWatcher {
```

Comment:
Now that this is in a `watcher` package, it makes sense to rename this simply `New` to prevent stutter. That way we can simply say `watcher.New(...)`.